### PR TITLE
Implemented a 5 day weather forecast application with openweathermap API

### PR DESCRIPTION
**Pull request by harrykeefe**
### Completed Stories

- [x] As a user, I would like to search for the current weather in my area
- [x] As a user, I would like to see the 5-day forecast
- [x] As a user, I would like to share the URL of the weather in my area and see the results
- [x] As a user, I would like to toggle between celsius and fahrenheit

### CODE REVIEW

**Comment by nethanelkohen** [comment]
Hey Harry, thanks for completing our take home assignment! I'm Nate from the engineering team at Contently. I'm going to drop a few comments in your PR. Looking forward to hearing back from you.

**Comment by nethanelkohen** [comment]
Why did you include `src/city.list.json`? Is it being used somewhere?

**Comment by nethanelkohen** [comment]
Why did you choose to search via zip code over city?

File: src/utils.js [hunk 1]
```diff
@@ -1,5 +1,102 @@
...
+              if (nJan1!==4) target.setMonth(0, 1 + ((4-nJan1)+7)%7);
+              return zeroPad(1 + Math.ceil((n1stThu-target)/604800000), 2);
+            })(),
+      '%w': nDay,
+      '%x': date.toLocaleDateString(),
+      '%X': date.toLocaleTimeString(),
+      '%y': (nYear + '').slice(2),
+      '%Y': nYear,
+      '%z': date.toTimeString().replace(/.+GMT([+-]\d+).+/, '$1'),
+      '%Z': date.toTimeString().replace(/.+\((.+?)\)$/, '$1')
```
Comment (nethanelkohen) [inline src/utils.js:93]: do you need all these extra functions?

**Comment by nethanelkohen** [comment]
Why did you decide against using state management? Even something simple like the Context API?

**Comment by nethanelkohen** [comment]
How would you improve the UX for users with 3G or slower connections?

**Comment by harrykeefe** [comment]
Hi Nathan, I appreciate you taking the time to review my code. I’ll respond to comments in order below. Hopefully this provides some clarity.

#### Why did you include src/city.list.json? Is it being used somewhere?

I originally was going to use this file to do a simple regex match filter so that the user could find proper city token. The openweathermap API does not provide any fuzzy matching so it is likely that the user input will not match the exact city ID required to get a response. Unfortunately, the file is rather large and for performance sake I opted against this implementation. I would remove this file before merging with production.

#### Why did you choose to search via zip code over city?

Similar to the answer above, I decided there would be less variance in the zip code input. I believe that until a more robust search function can be implemented this gives the user the highest likelihood of a successful match to their location.

#### do you need all these extra functions?

Good catch here, I could have gotten the same result with getDay() with less complexity. Outside of strftime, each of these other functions is used in the implementation of one of the stories.

#### Why did you decide against using state management? Even something simple like the Context API?

This is my first experience with React and I wanted to keep the number of APIs I had to learn to a minimum. If I were to recreate this application I would use Redux for state management. I am familiar with both VueX and the event bus pattern so I think this transition would be a smooth one. 

#### How would you improve the UX for users with 3G or slower connections?

My first course of action would be to deploy the application on a staging server and conduct a set of benchmarks. I would look for slow loading assets and ensure that my everything is being properly compressed (and thumbprinted to ensure the bundle is properly cached). Additionally, I would ensure that my components and supporting code is being properly lazy loaded. This could require creating a second bundle that could be shipped after the first with less runtime critical code. I would consider using a CDN to ensure performance across all regions. Depending on the performance demands and speed of the application I would also consider creating a fallback HTML version of the site. This would be a last resort as that would significantly increase development complexity.

**Pull request closed without merging**